File: TrelloApp/src/app/core/models/board.ts
import {Label} from './label';
import {UserBoard} from './user-board';
import {List} from './list';

export interface Board {
  id: number,
  title: string,
  background: string,
  createdAt: string,
  updatedAt?: string,
  lists: List[],
  userBoards: UserBoard[],
  labels: Label[]
}

export interface AddBoard {
  title: string;
  background: string;
}

export interface UpdateBoard {
  title?: string;
  background?: string;
}

File: TrelloApi/TrelloApi/src/Domain/Entities/Board.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TrelloApi.Domain.Entities;

[Table("Board")]
public class Board: Entity
{
    [StringLength(32), Required]
    public string Title { get; set; }

    [StringLength(32), Required]
    public string Background { get; set; }
    
    public ICollection<List> Lists { get; set; }

    public ICollection<UserBoard> UserBoards { get; set; }

    public ICollection<Label> Labels { get; set; }

    public Board(string title, string background)
    {
        Title = title;
        Background = background;
        CreatedAt = DateTime.UtcNow;
        UpdatedAt = null;
    }
}